### DISCUSSION TITLE
Sort Lines

### DISCUSSION THREAD

**Discussion by marlonjames71** [notaapp/talk#367] [Feature Requests]
Nota already has the feature where you can format text selection to be Title Case, Sentence case, UPPER CASE, and lower case which is fantastic.

To compliment that, I think another great formatting feature would be sorting lines. **Here are some examples:**

**Plain new lines**
| Before Formatting | After Formatting (Ascending) |
|-|-|
| python<br/>swift<br/>Rust | python<br/>Rust<br/>swift |

**Would Work with lists**
| Before Formatting | After Formatting (Descending) |
|-|-|
| • React<br/>• JavaScript<br/>• Typescript | • Typescript<br/>• React<br/>• JavaScript |

**Would Work with numbers**
| Before Formatting | After Formatting (Ascending) |
|-|-|
| 3<br/>5<br/>2 | 2<br/>3<br/>5 |

---

### Flow
1. User selects some lines of text
2. Bring up the command palette (cmd + shift + P)
3. Type in "sort" -> results show two options: Sort (Ascending) & Sort (Descending)
4. Hit enter -> the selected lines get assorted in place

### Bonus
Sorting within the same line (Same Line Sort <- I made that up).

Not expecting this, and this might be a lot trickier, but if sorting was allowed within the same line for selected text, that could be pretty powerful as well.

**For instance, take this sentence:**
"For the media items, the clips should be able to be filtered on with these filters: **person, dog, alarm, package**."

**Let's say the user selected this part of the sentence above:  _"person, dog, alarm, package"_. The user would then be able to pull up the command palette and look for Sort (Ascending), for example, and you'd end up with a sentence that looks like this:**
"For the media items, the clips should be able to be filtered on with these filters: alarm, dog, package, person."

<br/><br/>That's it. Thanks for taking the time to read this. Cheers!

**Comment by astoilkov** [comment]
We've discussed this feature in the past and we like it. It's a good addition that isn't hard and adds minimal complexity.

**Reply by marlonjames71** [reply]
Oh nice. Yeah I didn't think it would be that hard. I mean, I don't know your tech stack or your codebase or anything. But compared to the other stuff you guys have already done, I would think this would be on the easier and simpler side of things.
